engine/src/main/java/org/kigalisim/engine/number/EngineNumber.java

```diff
@@ -17,6 +17,10 @@
  * <p>This class encapsulates a numeric value (using BigDecimal for precision) and its
  * associated units string. It provides methods to access the value, units, and check
  * for specific unit types.</p>
+ * 
+ * <p>Key difference from JavaScript implementation: Uses BigDecimal instead of JavaScript's
```

Comment:
Thank you for this! However, I think I will document this elsewhere. Please remove this new paragraph (`<p>`)